### DISCUSSION TITLE
zpool compatibility setting

### DISCUSSION THREAD

**Discussion by pasha-19** [zbm-dev/zfsbootmenu#697] [Questions] [answered]
I may not be able to find the obvious but here goes.  

Sorry just noticed this page is EOL.
[https://docs.zfsbootmenu.org/en/v2.2.x/guides/debian/bookworm-uefi.html](url)

These same or similar notes appear in these links too:
[https://docs.zfsbootmenu.org/en/v2.3.x/guides/ubuntu/jammy-uefi.html#create-the-zpool](url)
[https://docs.zfsbootmenu.org/en/v2.3.x/guides/ubuntu/noble-uefi.html#create-the-zpool](url)

> The option -o compatibility=openzfs-2.1-linux ensures that the pool is created only with feature flags supported by the current ZFSBootMenu binary release. If you plan on building a custom ZFSBootMenu image that you will keep synchronized with your host, the compatibility option may be omitted.

> Binary releases of ZFSBootMenu are generally built with the latest stable version of ZFS. Future releases of ZFSBootMenu may therefore support newer feature sets. Check project release notes prior to updating or removing compatibility options and upgrading your system pool.

I searched for this in what I can guess might be "project release notes".

I admit an unlikely source for the complete project release notes including that it may not be the most recent release.

[https://docs.zfsbootmenu.org/en/v2.2.x/CHANGELOG.html](url)

This more current link was also checked.

[https://docs.zfsbootmenu.org/en/v2.3.x/CHANGELOG.html](url)

In both of the above there is one instance of the word compatibility and it does not relate to this situation.

This seemed to be the better choice.

[https://github.com/zbm-dev/zfsbootmenu/releases](url)

The word 'compatibility' does not appear based on a find on this page.  

Is there some place I have not found that contains this information?  Maybe my search was faulty.

Thanks

**Comment by bjoernwuest** [comment]
Not sure what is your concern, but on the github release notes for v2.2.1 it can be read that ZFS was upgraded to v2.2 - which actually is the newest ZFS version. v2.3 is still in development, yet I am looking forward it since you shall be able to grow RAIDZx-pools from then onwards.

**Reply by dshuman52** [reply]
The reference to 2.2 and 2.3 in the links above are the zfsbootmenu versions not the zfs package version, unfortunately they overlap.  My issue is in reading the note above should I choose to use zfsbootmenu, where in the zfsbootmenu documentation (in this case identified by the note as the project release notes) will I be informed if the  ' compatibility=openzfs-2.1-linux' should be upgraded.  Then I will be attempting to determine how to do this.  At this time, I do not believe I can determine when such an action might be necessary, which generates concerns about using the software.

**Reply by zdykstra** [reply, marked as answer]
Leave compatibility at 2.1 unless you have a specific reason to upgrade your zpool version. If you have a specific reason (e.g. you've read the ZFS documentation feature matrix), then you can upgrade your pool feature set.

Future releases of ZBM will more clearly outline what specific version of ZFS is used in the binary releases.
